### ISSUE TITLE
[sklearn#34252] RFC Automatically close pull requests linked to issues that are not ready for contributions

### ISSUE THREAD

**Issue by godofecht** [flooooooooooow/flow#344]
**Mirrored from [scikit-learn#34252](https://github.com/scikit-learn/scikit-learn/issues/34252)**

Original title: RFC Automatically close pull requests linked to issues that are not ready for contributions
Original labels: workflow
Created: 2026-06-11T14:49:55Z

---

Disclosure: I used LLM to proofread earlier drafts of this issue. All mistakes are due to me.

TL;DR: This RFC discusses whether pull requests linked to issues not yet ready for contributions should be automatically closed to reduce repetitive reviewer workload.

### The Situation

In the current review workflow, reviewers typically start with the following steps before inspecting the changes:

1. Manually check whether the pull request is linked to an issue that is ready for contributions (e.g., not labelled "Needs Triage").
2. If the linked issue is not yet ready for a pull request, reviewers need to explain this to the author and then close the pull request.

While this process is straightforward and the decision-making is usually quick, it can become repetitive over time and contribute to reviewer fatigue, increasing the maintenance burden.

#33521 reduces the time needed for step 1. However, explaining to contributors why a pull request is being closed because the linked issue is not ready (including wording and tone) is still time-consuming, and is often repeated in very similar forms. See https://github.com/scikit-learn/scikit-learn/pull/32951#pullrequestreview-3613580879, https://github.com/scikit-learn/scikit-learn/pull/33455#pullrequestreview-3894730435, and https://github.com/scikit-learn/scikit-learn/pull/33695#pullrequestreview-4060391909 for examples.

### Proposed Solution

During the 2026 sprint, we (@AnneBeyer, @StefanieSenger, and myself) discussed mitigating step 2 by introducing a bot that automatically closes pull requests linked to issues that are not yet ready for contributions. This builds on #33521, and is relatively straightforward from a technical standpoint.

However, the proposed approach has at least two potential concerns:

1. It may close pull requests that are intended as PoC for issues still under discussion, such as #34242.
2. It may introduce security considerations (e.g., GitHub token handling).

Despite these concerns, we believe this RFC could help address these issues and improve the overall review experience in the long run.

---

## Flow context

This issue is mirrored from scikit-learn to explore whether Flow encounters the same problem or can approach it differently. Add analysis below.

**Comment by godofecht** [comment]
Moved to https://github.com/godofecht/flow-scikit/issues/79 in the flow-scikit repository. Further library-specific triage and implementation will happen there.
